com.github.dariasteam.cowsrevenge.json

```diff
@@ -0,0 +1,31 @@
+{
+  "app-id": "com.github.dariasteam.cowsrevenge",
+  "branch": "stable",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "1.6",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "cowsrevenge",
+  "finish-args": [
+     "--socket=x11",
+     "--socket=pulseaudio",
+     "--persist=.godot",
+     "--share=ipc",
+     "--device=dri"
+  ],
+  "build-options" : {
+    "split" : false,
+    "no-debuginfo" : true
+  },
+  "modules": [
+    {
+      "name": "cowsrevenge",
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/Dariasteam/Cows-Revenge-Flatpak",
```

Comment:
The sources should be either here or upstream, I'm not sure I understand having an intermediate repo. Icons, desktop files, and appdata are not Flatpak specific and would be reused for all packages.

Reply:
The idea of the intermediate repo is to separate the completely free source code and art stuff from the binary. My original intention was to create a makefile that downloads the full godot sources, compiles it and then generates the game itself (similar to what f-droid use) but I had some problems and decided to do this weird thing. Also I wrote the metadata files by the way when reading the flatpak needs so the result is this mess. Should I retry the engine compiling method and use a single repository?

Reply:
> My original intention was to create a makefile that downloads the full godot sources, compiles it and then generates the game itself 

Yes that is the ideal solution.

Here is an example of building Godot: https://github.com/flathub/org.godotengine.Godot/blob/master/org.godotengine.Godot.json#L34-L72

Reply:
Ok, thank you for the example

Reply:
I was wondering if we could import godot as a baseapp and then remove it afterwards, in a manner similar to an SDK - that would make it a bit quicker to build - but that's more idle musing as to an opimization, it wouldn't make a difference to the outcome.

Reply:
It looks like @TingPing is already in it? https://github.com/godotengine/godot/issues/5597

Reply:
Question: Is the build process of flatpak-builder forbidding to acces access internet? This [makefile](https://github.com/Dariasteam/Cows-Revenge/blob/stable/Makefile) successfuly download godot sources when called manually but returns `fatal: unable to access 'https://github.com/godotengine/godot/': Couldn't resolve host 'github.com'
` when called by flatpak-builder

Reply:
>  Is the build process of flatpak-builder forbidding to acces access internet?

Yes. All sources should be downloaded by `flatpak-builder`.

Reply:
I'm having some problems with this. In order to export godot requires to put some templates files in ~/.local/share/godot and apparently theres no way to specify other path using flags. How should I deal with this considering flatpak is not allowing me to do stuff in ~/ (I have already tried "using --persist=.local/share/godot" but it looks only useful when running not compiling)

Reply:
> In order to export godot requires to put some templates files in ~/.local/share/godot and apparently theres no way to specify other path using flags.

Honestly the best solution is just to write a patch fixing that in godot.

Reply:
Ok, it can be fixed using [unix_global_settings_path](https://github.com/godotengine/godot-docs/blob/master/development/compiling/packaging_godot.rst), it is also used in your example. Sorry.